lib/ddtrace/contrib/mysql2/instrumentation.rb

```diff
@@ -19,15 +19,17 @@ def self.included(base)
         module InstanceMethods
           def query(sql, options = {})
             Datadog::Tracing.trace(Ext::SPAN_QUERY) do |span|
+              service = Datadog::Tracing.configuration_for(self, :service_name) || datadog_configuration[:service_name]
+
               span.resource = sql
-              span.service = datadog_pin.service
+              span.service = service
```

Comment:
Minor: Perhaps we could extract the service and pass it in to `Tracing.trace`, rather than mutating it after-the-fact?

Reply:
It doesn't matter too much either way, we don't have consistent application of this.

Reply:
I normally add it directly to `#trace` as well, but no strong opinions here at this moment.